pilot/pkg/security/authz/matcher/string.go

```diff
@@ -48,6 +48,14 @@ func StringMatcherWithPrefix(v, prefix string) *matcherpb.StringMatcher {
 	// the Envoy StringMatcher doesn't allow empty prefix/suffix.
 	case v == "*":
 		return StringMatcherRegex(".+")
+	case strings.HasPrefix(v, "regex:"):
+		return &matcherpb.StringMatcher{
+			MatchPattern: &matcherpb.StringMatcher_SafeRegex{
+				SafeRegex: &matcherpb.RegexMatcher{
+					Regex: strings.TrimPrefix(v, "regex:"),
```

Comment:
you need to take care of the `prefix` argument, see how it is handled in other cases.